### PR TITLE
TPD-1267 Update Payment Method API, TPD-1304 Process One Time Gift Authenticated, TPD-1173 Donor Covers fee, TPD-1375 Pledge Summary API

### PR DESCRIPTION
**Pull request by jason-rice**
…hecks, created update payment method action method in controller and the method in GivingPaymentModel
[touchpointsoftware.atlassian.net/browse/TPD-1267](https://touchpointsoftware.atlassian.net/browse/TPD-1267)
[touchpointsoftware.atlassian.net/browse/TPD-1304](https://touchpointsoftware.atlassian.net/browse/TPD-1304)
[touchpointsoftware.atlassian.net/browse/TPD-1173](https://touchpointsoftware.atlassian.net/browse/TPD-1173)
[touchpointsoftware.atlassian.net/browse/TPD-1375](https://touchpointsoftware.atlassian.net/browse/TPD-1375)

### CODE REVIEW

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs [hunk 1]
```diff
@@ -102,7 +98,7 @@ public Message CreateMethod(GivingPaymentViewModel viewModel)
                         return Message.createErrorReturn("Bank account routing number is required.", Message.API_ERROR_PAYMENT_METHOD_REQUIRED_FIELD_EMPTY);
                     }
                 }
-                else
+                else if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Amex || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Discover || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.MasterCard || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Visa)
```
Comment (brondavies) [inline CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs:101]: use `PaymentMethodTypeCode.IsCard()`

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs [hunk 2]
```diff
@@ -171,10 +171,12 @@ public Message CreateMethod(GivingPaymentViewModel viewModel)
                 {
                     bankValidation = PaymentValidator.ValidateBankAccountInfo(viewModel.bankInfo.accountNumber, viewModel.bankInfo.routingNumber);
                     if (bankValidation.error != 0)
+                    {
                         return Message.createErrorReturn(bankValidation.data, bankValidation.error);
+                    }
                 }
             }
-            else
+            else if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Amex || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Discover || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.MasterCard || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Visa)
```
Comment (brondavies) [inline CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs:179]: use `PaymentMethodTypeCode.IsCard()`

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs [hunk 3]
```diff
@@ -204,19 +206,21 @@ public Message CreateMethod(GivingPaymentViewModel viewModel)
...
-            paymentMethod.GatewayAccountId = account.GatewayAccountId;
             var gateway = CurrentDatabase.Gateway(viewModel.testing, account, PaymentProcessTypes.RecurringGiving);
+            paymentMethod.GatewayAccountId = account.GatewayAccountId;
 
             if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Bank)
             {
                 gateway.StoreInVault(paymentMethod, PaymentType.Ach, viewModel.bankInfo, null, viewModel.testing);
             }
-            else
+            else if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Amex || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Discover || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.MasterCard || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Visa)
```
Comment (brondavies) [inline CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs:223]: same here

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs [hunk 4]
```diff
@@ -232,31 +236,252 @@ public Message CreateMethod(GivingPaymentViewModel viewModel)
...
+                    if (!viewModel.bankInfo.accountNumber.HasValue())
+                    {
+                        return Message.createErrorReturn("Bank account number is required.", Message.API_ERROR_PAYMENT_METHOD_REQUIRED_FIELD_EMPTY);
+                    }
+                    if (!viewModel.bankInfo.routingNumber.HasValue())
+                    {
+                        return Message.createErrorReturn("Bank account routing number is required.", Message.API_ERROR_PAYMENT_METHOD_REQUIRED_FIELD_EMPTY);
+                    }
+                }
+                else if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Amex || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Discover || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.MasterCard || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Visa)
```
Comment (brondavies) [inline CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs:325]: same here

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs [hunk 5]
```diff
@@ -668,7 +895,7 @@ public Message ProcessOneTimePayment(GivingPaymentViewModel viewModel)
                     return Message.createErrorReturn("Bank account routing number is required.", Message.API_ERROR_PAYMENT_METHOD_REQUIRED_FIELD_EMPTY);
                 }
             }
-            else
+            else if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Amex || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Discover || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.MasterCard || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Visa)
```
Comment (brondavies) [inline CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs:898]: same here

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs [hunk 6]
```diff
@@ -803,10 +1037,10 @@ public Message ProcessOneTimePayment(GivingPaymentViewModel viewModel)
                     return Message.createErrorReturn(transactionResponseForBankPayment.Message, Message.API_ERROR_BANK_PAYMENT_FAILED);
                 }
             }
-            else
+            else if (viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Amex || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Discover || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.MasterCard || viewModel.paymentTypeId == CmsData.Codes.PaymentMethodTypeCode.Visa)
```
Comment (brondavies) [inline CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs:1040]: same here

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs (hunk 1 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs]: fixed this

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs (hunk 2 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs]: fixed this

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs (hunk 3 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs]: fixed this

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs (hunk 5 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs]: fixed this

File: CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs (hunk 6 above)
Comment (jason-rice) [reply CmsWeb/Areas/Giving/Models/GivingPayment/GivingPaymentModel.cs]: fixed this

File: CmsData/Migrations/20200702_Donor_Covers_Fee.sql [hunk 7]
```diff
@@ -30,7 +30,7 @@ BEGIN
 INSERT INTO [dbo].SettingMetadata
            ([SettingId], [DisplayName], [Description], [DataType], [SettingCategoryId], [DefaultValue])
      VALUES
-           ('DonorCoversFeeOnByDefualt', NULL, 'This setting makes the check box for donor covers fees checked by default.', 5, 3, NULL)
+           ('DonorCoversFeeOnByDefualt', NULL, 'This setting makes the check box for donor covers fees checked by default.', 1, 3, NULL)
```
Comment (brondavies) [inline CmsData/Migrations/20200702_Donor_Covers_Fee.sql:33]: Typo: DonorCoversFeeOnByDef**au**lt

Comment (jason-rice) [reply CmsData/Migrations/20200702_Donor_Covers_Fee.sql:33]: fixed this

**Pull request merged**